### PR TITLE
[Feature][Style] Configure Spotless to format YAML file in DS

### LINKED ISSUE #11699: [Feature][Style] Configure Spotless to format YAML file in DS
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Configure Spotless to format YAML file code in DS.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jieguangzhou**
## Purpose of the pull request

close: #11699

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

### CODE REVIEW

**Review by EricGao888 — APPROVED** [review]
LGTM

File: pom.xml [hunk 1]
```diff
@@ -684,6 +684,16 @@
                         </includes>
                         <flexmark />
                     </markdown>
+                    <formats>
+                        <format>
+                            <includes>
+                                <include>**/*.yaml</include>
+                            </includes>
+                            <prettier>
+                                <prettierVersion>2.0.5</prettierVersion>
```
Comment (SbloodyS) [inline pom.xml:693]: Can you provide several different types of yaml files to see the effect after formatting?

As far as I know, we have many different types of yaml files. Such as `docker-compose yaml` and `spring application yaml`.

Comment (jieguangzhou) [reply pom.xml:693]: > Can you provide several different types of yaml files to see the effect after formatting?
> 
> As far as I know, we have many different types of yaml files. Such as `docker-compose yaml` and `spring application yaml`.

I have just tested some of the other YAML files in DS and some of them (may be not a normal YAML file) are getting errors when formatting.

For example: deploy/kubernetes/dolphinscheduler/templates/configmap.yaml

I think we can't cannot add a formatted operation directly. 

WDYT @EricGao888 @SbloodyS

Comment (jieguangzhou) [reply pom.xml:693]: Or we can skip some folders and format only the ones we need

image

Comment (EricGao888) [reply pom.xml:693]: What errors did you get exactly?

Comment (jieguangzhou) [reply pom.xml:693]: Document is not valid YAML (bad indentation?)

```
[ERROR] Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:2.23.0:apply (default-cli) on project dolphinscheduler: Execution default-cli of goal com.diffplug.spotless:spotless-maven-plugin:2.23.0:apply failed: Unexpected response status code at /prettier/format [HTTP 501] -- (Error while formatting: SyntaxError: Document is not valid YAML (bad indentation?) (18:1)
[ERROR]   16 | #
[ERROR]   17 | {{- if .Values.common.fsFileResourcePersistence.enabled }}
[ERROR] > 18 | apiVersion: v1
[ERROR]      | ^^^^^^^^^^^^^^
[ERROR] > 19 | kind: PersistentVolumeClaim
[ERROR]      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] > 20 | metadata:
[ERROR]      | ^
[ERROR]   21 |   name: {{ include "dolphinscheduler.fullname" . }}-fs-file
[ERROR]   22 |   labels:
[ERROR]   23 |     app.kubernetes.io/name: {{ include "dolphinscheduler.fullname" . }}-fs-file)
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

Comment (jieguangzhou) [reply pom.xml:693]: I have closed this PR, because I think we need to discuss how to format the YAML file first.

Comment (EricGao888) [reply pom.xml:693]: Okay, in that case, I added an `discussion` label to #11699 .

**Pull request closed without merging**